celery/result.py

```diff
@@ -101,6 +101,17 @@ def __init__(self, id, backend=None,
         self.parent = parent
         self.on_ready = promise(self._on_fulfilled)
         self._cache = None
+        self._ignored = False
+
+    @property
+    def ignored(self):
+        if hasattr(self, '_ignored'):
```

Comment:
In which case we won't have `_ignored` if it's initialized in the constructor.

Reply:
I introduced this check in order to account for sub-classes with overridden constructor, please see my comment above.